cocotb/share/lib/vpi/VpiCbHdl.cpp

```diff
@@ -671,7 +648,10 @@ VpiIterator::VpiIterator(GpiImplInterface *impl, GpiObjHdl *hdl) : GpiIterator(i
     vpiHandle vpi_hdl = m_parent->get_handle<vpiHandle>();
 
     int type = vpi_get(vpiType, vpi_hdl);
-    if (NULL == (selected = iterate_over.get_options(type))) {
+    try {
+        selected = &iterate_over.at(type);
+    }
+    catch (std::out_of_range const&) {
```

Comment:
```suggestion
    if (iterate_over.end() == (selected = iterate_over.find(type))) {
```

Reply:
Doesn't work because `find` returns a `pair`. I can write the more verbose version, but my feeling was that in the long run this constructor should probably throw anyway if the type doesn't exist, rather than allowing an invalid object to be constructed.

Reply:
The alternative is probably to add back a helper like:
```C++
template<typename K, typename V>
V* get_or_null(std::map<K, V>& m, K const& k) {
    auto it = m.find(k);
    return it == m.end() ? nullptr : &it->second;
}
```
which is still a lot lighter weight than the whole class.

Reply:
>`find` returns a `pair`.

What? https://en.cppreference.com/w/cpp/container/map/find. Or are you talking about std::multimap::equal_range which does return a pair?

And you are correct that we should throw eventually, but I feel that perhaps we can throw our own more descriptive exception than a standard exception.

Reply:
Sorry, I meant "returns an `iterator<pair<K, V>>`". The takeaway is the same though, `selected` needs to be a `K*`, like in my `get_or_null` function.

> And you are correct that we should throw eventually, but I feel that perhaps we can throw our own more descriptive exception than a standard exception.

Indeed, which is why for now I'm discarding the exception rather than re-reraising it.

Reply:
I guess it really should never trigger, so whatever. It's easy enough to refactor in the future if necessary.

Reply:
A good [validation](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#e2-throw-an-exception-to-signal-that-a-function-cant-perform-its-assigned-task) of my heartache with this. I'm still willing to bring this in. Its a very Pythonic solution, but this is C++.